kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyInputFormat.java

```diff
@@ -227,6 +226,7 @@ public Configuration getConf() {
   @Override
   public void setConf(Configuration configuration) {
     conf = configuration;
+    DefaultConfiguration.set(conf);
```

Comment:
It's a fair question. The way that Kite passes around DefaultConfiguration is in general a little scary. In this case we are safe. The `setConf()` method will be called when the `DatasetKeyInputFormat` instance is created by the MapReduce framework. That happens before any user code is run and it only creates a single instance per JVM. If the user had to override the DefaultConfiguration later, they'd be free to do so and they'd be the only ones touching it at that point.